test/test_torch.py

```diff
@@ -2519,6 +2526,57 @@ def test_serialization(self):
             self.assertEqual(c[1], c[3], 0)
             self.assertEqual(c[4], c[5][1:4], 0)
 
+            # check that serializing the same storage view object unpickles
+            # it as one object not two (and vice versa)
+            views = c[7]
+            self.assertEqual(views[0]._cdata, views[1]._cdata)
+            self.assertEqual(views[0], views[2])
+            self.assertNotEqual(views[0]._cdata, views[2]._cdata)
```

Comment:
I mean that instead of comparing the `_cdata` values, we should compare actualy Python objects. If you serialized the same Python storage twice, it should also get deserialized as the exact same object (so that the `is` relation is maintained). I'd change `self.assertEqual(views[0]._cdata, views[1]._cdata)` to `self.assertIs(views[0], views[1])`.